src/QsCompiler/LlvmBindings/Interop/LLVMExecutionEngineRef.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) .NET Foundation and Contributors. All Rights Reserved. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace LlvmBindings.Interop
+{
+    public unsafe partial struct LLVMExecutionEngineRef : IDisposable, IEquatable<LLVMExecutionEngineRef>
+    {
+        public IntPtr Handle;
+
+        public LLVMExecutionEngineRef(IntPtr handle)
+        {
+            this.Handle = handle;
+        }
+
+        public LLVMTargetDataRef TargetData => (this.Handle != IntPtr.Zero) ? LLVM.GetExecutionEngineTargetData(this) : default;
+
+        public LLVMTargetMachineRef TargetMachine => (this.Handle != IntPtr.Zero) ? LLVM.GetExecutionEngineTargetMachine(this) : default;
+
+        public static implicit operator LLVMExecutionEngineRef(LLVMOpaqueExecutionEngine* value) => new LLVMExecutionEngineRef((IntPtr)value);
+
+        public static implicit operator LLVMOpaqueExecutionEngine*(LLVMExecutionEngineRef value) => (LLVMOpaqueExecutionEngine*)value.Handle;
+
+        public static bool operator ==(LLVMExecutionEngineRef left, LLVMExecutionEngineRef right) => left.Handle == right.Handle;
+
+        public static bool operator !=(LLVMExecutionEngineRef left, LLVMExecutionEngineRef right) => !(left == right);
+
+        public void AddGlobalMapping(LLVMValueRef global, IntPtr addr) => LLVM.AddGlobalMapping(this, global, (void*)addr);
+
+        public void AddModule(LLVMModuleRef m) => LLVM.AddModule(this, m);
+
+        public void Dispose()
```

Comment:
Keeping unmanaged handles may require a more careful disposing and destructor. https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
Filed an issue for follow up at https://github.com/microsoft/qsharp-compiler/issues/1324.